flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/table/RowDataToAttributeValueConverter.java

```diff
@@ -0,0 +1,72 @@
+package org.apache.flink.connector.dynamodb.table;
+
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.conversion.DataStructureConverters;
+import org.apache.flink.table.types.DataType;
+
+import software.amazon.awssdk.enhanced.dynamodb.TableSchema;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.StaticTableSchema;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.flink.table.data.RowData.createFieldGetter;
+
+/** Converts from {@link RowData} to {@link AttributeValue}. */
+public class RowDataToAttributeValueConverter implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final DataType physicalDataType;
+    private transient TableSchema<RowData> tableSchema;
+
+    public RowDataToAttributeValueConverter(DataType physicalDataType) {
+        this.physicalDataType = physicalDataType;
+        this.tableSchema = createTableSchema();
+    }
+
+    private StaticTableSchema<RowData> createTableSchema() {
+        List<DataTypes.Field> fields = DataType.getFields(physicalDataType);
+        StaticTableSchema.Builder<RowData> builder = TableSchema.builder(RowData.class);
+        for (int i = 0; i < fields.size(); i++) {
+            DataTypes.Field field = fields.get(i);
+            RowData.FieldGetter fieldGetter =
+                    createFieldGetter(field.getDataType().getLogicalType(), i);
+
+            builder =
+                    addAttribute(
+                            builder, field.getDataType().getConversionClass(), field, fieldGetter);
+        }
+        return builder.build();
+    }
+
+    private <T> StaticTableSchema.Builder<RowData> addAttribute(
+            StaticTableSchema.Builder<RowData> builder,
+            Class<T> convertedClass,
+            DataTypes.Field field,
+            RowData.FieldGetter fieldGetter) {
+
+        return builder.addAttribute(
+                convertedClass,
+                a ->
+                        a.name(field.getName())
+                                .getter(
+                                        rowData ->
+                                                (T)
+                                                        DataStructureConverters.getConverter(
+                                                                        field.getDataType())
+                                                                .toExternal(
+                                                                        fieldGetter.getFieldOrNull(
+                                                                                rowData)))
+                                .setter(((rowData, t) -> {})));
+    }
+
+    public Map<String, AttributeValue> convertRowData(RowData row) {
```

Comment:
nit: Public methods should typically go above private, for readability

Reply:
done